src/main/scala/beam/router/BeamRouter.scala

```diff
@@ -211,6 +214,7 @@ class BeamRouter(
         sendWorkTo(worker, work, originalSender, receivePath = "GimmeWork")
       }
     case routingResp: RoutingResponse =>
+      eventsManager.processEvent(RoutingResponseEvent(routingResp))
```

Comment:
Is there any noticeable overhead from this?

Reply:
Good point. I don't think it will, but I can run prod run before actual merge.